### ISSUE TITLE
Window won't remember the dimension i set and reset to minimum

### ISSUE THREAD

**Issue by Tetractys** [wwmm/easyeffects#2798]
### EasyEffects Version

7.1.3-1

### What package are you using?

Arch (easyeffects)

### Distribution

Arch Linux

### Describe the bug

After issuing in cli the command:

'gsettings reset-recursively com.github.wwmm.easyeffects'

to avoid the crashes during tab changes or netflix page reload (using Opera Browser - chromium based), if i modify the windows dimension to see everything inside (especially the graphic equalizer) nextime i start easyeffects it create the minimum possible window dimension forgotting/losing my dimensions modifications. See screenshot:

![image_2023-12-10_104151148](https://github.com/wwmm/easyeffects/assets/465836/7a2f36ad-f1cd-4388-9453-b6e1ef0e94df)

Any clue please?

### Expected Behavior

when dimensioned the windows should remember this:

![image_2023-12-10_104327693](https://github.com/wwmm/easyeffects/assets/465836/2097ffe1-c91c-4921-b2d3-18508a91d6ce)

### Debug Log

<details><summary>Debug Log</summary><br/>

```
Paste your log here
```
</details>

**Comment by wwmm** [comment]
>gsettings reset-recursively com.github.wwmm.easyeffects

This will also reset the saved values for the window's dimension.

**Comment by wwmm** [comment]
If your system is killing EasyEffects the real fix would be to figure out why. What unfortunately isn't always easy. Does this happens only when a specific plugin is used?

**Comment by Tetractys** [comment]
> > gsettings reset-recursively com.github.wwmm.easyeffects
> 
> This will also reset the saved values for the window's dimension.

How do i set the dimension i want remain stable? redimensioning the window does nothing. Any time i restart EasyEffects the window is on its defaults...

**Comment by Tetractys** [comment]
> If your system is killing EasyEffects the real fix would be to figure out why. What unfortunately isn't always easy. Does this happens only when a specific plugin is used?

This problem was fixed with the solution as reported in issue #2298

Note: I'm on Arch with a clean Cinnamon 6.0.0 installation. Pipewire, pipewire-pulse and wireplumber working as intended.

**Comment by Tetractys** [comment]
i solved temporary with dconf editor and xdotool.
I setup the window width and height so when it opens the window itself is correct in height and width.

**[Feature Request]** A switch that permit to maintain the position and dimension of the easyeffects window.
Before the migration to GTK4 all was remembered.

![image_2023-12-10_180034533](https://github.com/wwmm/easyeffects/assets/465836/57a62490-86a0-48db-9d36-2e01af2e8273)

**Comment by wwmm** [comment]
>Before the migration to GTK4 all was remembered.

And it still does it. I am also on Arch Linux and the window size is automatically restored just fine as long as the settings are not reset before launching the window again.

**Comment by wwmm** [comment]
This is where `width,height,maximized and fullscreen` status are restored. As long as gsettings (dconf) is working properly this is all restored. I wonder if dconf is somehow broken on your installation. But in this case other settings would also be broken...

**Comment by wwmm** [comment]
Or maybe gtk4 is misbehaving outside of gnome...

**Comment by violetmage** [comment]
@Tetractys you're using X11, right?

@wwmm Perhaps it is some difference in the way gtk4 resizes windows on wayland and x11

**Comment by wwmm** [comment]
>Perhaps it is some difference in the way gtk4 resizes windows on wayland and x11

I don't know. I will try a quick test on X11.

**Comment by wwmm** [comment]
>Perhaps it is some difference in the way gtk4 resizes windows on wayland and x11

No. It is not this. I logged on a gnome X11 session and the restoration worked fine just like on wayland. Either it is working only on GNOME or something is broken in the systems where it fails.

**Comment by Tetractys** [comment]
@wwmm @violetmage 
Cinnamon 6 has an experimental build for wayland i'll try it. thanks a lot for the moment!

Q: what is the process or service that monitor the windows modifications and store it for GTK4?

**Comment by wwmm** [comment]
>what is/are the process/es or service/s that monitor the windows modifications and store it for GTK4?

EasyEffects uses a Gsettings (dconf) database for this. The width/height keys you manually set were put there exactly for the restoration task. That is why is weird it is not working for you. WE store all parameters in gsettings. It does not make sense that only these twos are being erased.

Kill EE `easyeffects -q` and restart it i ndebuge mode `G_MESSAGES_DEBUG=easyeffects easyeffects` maybe there is some kind of gtk/gsettings warning that will help to understand what is going on.

**Comment by wwmm** [comment]
>what is/are the process/es or service/s that monitor the windows modifications and store it for GTK4?

We save the width and height in the gsettings database whenever our window is closed https://github.com/wwmm/easyeffects/blob/41d1720a3e381b6474e2192a5b00b1fed975c126/src/application_ui.cpp#L220. If you keep dconf-editor window opened you should see these parameters value being updated whenever you close EE window and it has a different size than before.

**Comment by Tetractys** [comment]
Probably i have the settings.ini a bit weird.....

Here the startup with G_MESSAGES_DEBUG:

https://pastebin.com/vtQmTrnm

**Comment by servimo** [comment]
EasyEffects doesn't have a key "remember-position" in gsettings.

Em seg., 11 de dez. de 2023 12:30, Davide Monge ***@***.***>
escreveu:
